src/compas_view2/objects/arrowobject.py

```diff
@@ -1,11 +1,118 @@
-from compas.datastructures import Mesh
-from .meshobject import MeshObject
+from .bufferobject import BufferObject
 
 
-class ArrowObject(MeshObject):
+class ArrowObject(BufferObject):
     """Object for displaying COMPAS arrow geometry."""
 
-    def __init__(self, data, u=16, **kwargs):
-        super().__init__(Mesh.from_shape(data, u=u), **kwargs)
+    def __init__(self, data, u=16, color=None, facecolor=None, linecolor=None, **kwargs):
+        super().__init__(data, **kwargs)
         self._u = u
-        self._data = data
+        self.facecolor = facecolor or color
+        self.linecolor = linecolor or color
+        self._vertices = None
+        self._faces = None
+
+    @property
+    def vertices(self):
+        if not self._vertices:
+            self._data.to_vertices_and_faces(u=self._u)
+            self._vertices = self._data.vertices
+            self._faces = self._data.faces
+        return self._vertices
+
+    @property
+    def faces(self):
+        if not self._faces:
+            self._data.to_vertices_and_faces(u=self._u)
+            self._vertices = self._data.vertices
+            self._faces = self._data.faces
+        return self._faces
+
+    @classmethod
+    def from_other(cls, other, **kwargs):
+        arrow = cls(other._data, **kwargs)
+        arrow._vertices = other._vertices
+        arrow._faces = other._faces
+        return arrow
+
+    def _points_data(self):
+        color = self.pointcolor or self.default_color_points
+        positions = []
+        colors = []
+        elements = []
+        i = 0
+        for vertex in self.vertices:
+            positions.append(vertex)
+            colors.append(color)
+            elements.append([i])
+            i += 1
+        return positions, colors, elements
+
+    def _frontfaces_data(self):
```

Comment:
was the separation from MeshObject because of memory consumption concern related to face colors? If that's the cause maybe I should address this in MeshObject itself as well

Reply:
no actually now looking at it, it seems to be leftover from an attempt to make the arrow faster for visualisation in the animation of the collapse sequence of the venice bridge but it never gave the desired results...